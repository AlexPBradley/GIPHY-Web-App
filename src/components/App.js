import React, { useState } from "react";
import "../styles/app.css";
import Search from "./Search";
import SearchResults from "./SearchResults";
import Expanded from "./Expanded";
import logo from "../styles/images/giphy-logo.jpg";

const App = () => {
  const [gifResults, setGifResults] = useState([]);
  const [isSelected, setIsSelected] = useState(false);
  const [selected, setSelected] = useState();

  return (
    <div className="app">
      <img className="logo" src={logo} alt="giphy-logo" />
      <Search setGifResults={setGifResults} />
      {isSelected ? (
        <Expanded
          results={gifResults}
          selected={selected}
          setIsSelected={setIsSelected}
        />
      ) : null}
      <SearchResults
        results={gifResults}
        isSelected={setIsSelected}
        setSelected={setSelected}
      />
    </div>
  );
};

export default App;
