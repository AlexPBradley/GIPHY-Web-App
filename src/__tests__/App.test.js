import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<App />);
    
    expect(asFragment()).toMatchSnapshot();
  });

  it("logo renders", () => {
    render(<App />);
    
    const logo = screen.getByTestId("logo");

    expect(logo).toBeInstanceOf(HTMLImageElement);
    expect(logo).toHaveClass("logo");
  });

  it("page top renders", () => {
    render(<App />);

    const pageTop = screen.getByTestId("page-top");

    expect(pageTop).toBeInstanceOf(HTMLAnchorElement);
    expect(pageTop).toHaveClass("page-top");
  });
});
