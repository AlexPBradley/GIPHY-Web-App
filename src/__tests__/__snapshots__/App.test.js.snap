// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <img
      alt="giphy-logo"
      class="logo"
      data-testid="logo"
      src="giphy-logo.jpg"
    />
    <div
      class="search"
    >
      <form
        class="search-form"
        data-testid="search-form"
      >
        <input
          class="search-input"
          data-testid="search-input"
          placeholder="Search..."
          type="text"
        />
      </form>
      <div
        class="limitButtons"
      >
        <button
          class="search-button"
          type="button"
        >
          <span
            id="button-background"
            style="transform: translate(0px);"
          />
          10
        </button>
        <button
          class="search-button"
          id="middle"
          type="button"
        >
          25
        </button>
        <button
          class="search-button"
          type="button"
        >
          50
        </button>
      </div>
    </div>
    <h1>
      Trending Results
    </h1>
    <div
      class="search-results"
    />
    <a
      class="page-top"
      data-testid="page-top"
      href="#top"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-up "
        data-icon="arrow-up"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 384 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M214.6 41.4c-12.5-12.5-32.8-12.5-45.3 0l-160 160c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 141.2V448c0 17.7 14.3 32 32 32s32-14.3 32-32V141.2L329.4 246.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3l-160-160z"
          fill="currentColor"
        />
      </svg>
    </a>
  </div>
</DocumentFragment>
`;
